tests/corrgi/test_pcf.py

```diff
@@ -17,6 +17,23 @@ def test_pcf_counts_are_correct(
     npt.assert_allclose(expected_rr, pcf_rr_counts, rtol=2e-3)
 
 
+def test_pcf_counts_with_weights_are_correct(
+    dask_client,
+    pcf_gals_weight_catalog,
+    pcf_rans_weight_catalog,
+    pcf_dd_counts_with_weights,
+    pcf_rr_counts_with_weights,
+    pcf_params,
+):
+    proj_corr = ProjectedCorrelation(params=pcf_params, use_weights=True)
+    counts_dd, counts_rr = compute_autocorrelation_counts(
+        pcf_gals_weight_catalog, pcf_rans_weight_catalog, proj_corr
+    )
+    expected_dd, expected_rr = counts_dd.transpose([1, 0]), counts_rr.transpose([1, 0])
+    npt.assert_allclose(expected_dd, pcf_dd_counts_with_weights, rtol=1e-3)
+    npt.assert_allclose(expected_rr, pcf_rr_counts_with_weights, rtol=2e-3)
```

Comment:
This looks fine when the requested estimator is NAT (which requires DD & RR counts), but will not work for other estimators which requires cross-counts. I guess at some point there is the need for a method compute_cross_counts() which returns pairs from two different catalogs

Reply:
@samotracio Thank you, that is correct! In its current state corrgi is only supporting the natural estimator. I'll be addressing the remaining ones, as reported in https://github.com/lincc-frameworks-mask-incubator/corrgi/issues/23.